docs(a11y): add accessibility docs page

dancormier opened the pull request:
[STACKS-472](https://stackoverflow.atlassian.net/browse/STACKS-472)
This PR adds an accessibility page to our docs site.

> [!TIP]
> ~~In the process of creating this docs page, I made a few QoL improvements, most notably with the addition of an 11ty `stacks-markdown` plugin so we can write our docs pages using markdown when we see fit (this is distinct from our `markdown` plugin since it is intended accept raw markdown to output Stacks docs markup). There are also minor refactors of some partials and the heading plugin (`header.js`).~~
> These changes have been moved to [dcormier/docs-markdown-support branch](https://github.com/StackExchange/Stacks/tree/dcormier/docs-markdown-support)

> [!NOTE]
> I omitted the "known issues" section shown in the [figma mock](https://www.figma.com/file/3OlPL6x0chNbMIPjP2vxas/Stack-Overflow-3.0---Site-Modernization?node-id=3747%3A15709&mode=dev) since it only contained an entry for the button group component which was only relevant to the previous button group style.

---

Question for @CGuindon: Should we make any changes to the [Color fundamentals > Accessibility standards](https://deploy-preview-1670--stacks.netlify.app/product/base/color-fundamentals/#accessibility-standards) section?

---

### TODO

- [x] Update left nav
    - [x] See the [figma spec](https://www.figma.com/file/3OlPL6x0chNbMIPjP2vxas/Stack-Overflow-3.0---Site-Modernization?node-id=3747%3A15709&mode=dev)
    - [x] Move "Current color" to `BASE` section

dancormier committed:
docs(a11y): add accessibility docs page

dancormier committed:
Continue buildout of accessibility docs page

dancormier committed:
Replace color-swatch plugin with include

dancormier committed:
Merge branch 'develop' into STACKS-472/docs-a11y-page

dancormier committed:
Lots more enhancements!

dancormier committed:
Add remaining examples; cleanup

dancormier committed:
Apply custom height to example lines

dancormier committed:
Fix height on "Floating" examples

dancormier committed:
Merge branch 'develop' into STACKS-472/docs-a11y-page

dancormier commented on docs/plugins/stacks-markdown.js:
This file is a new plugin for Stacks documentation markdown. It converts the markdown input to HTML, modifying headings, paragraphs, and code elements with our standard Stacks documentation classes.

> [!NOTE]
> This plugin includes an option for `inline` markdown. This results in the `p` tags being replaced with `span`s so text can be placed in whatever container is most appropriate (usually a parent `p.stacks-copy` as seen in [accessibility.html#L128-L132](https://github.com/StackExchange/Stacks/blob/STACKS-472/docs-a11y-page/docs/product/guidelines/accessibility.html#L128-L132)), allowing us to use other common markdown syntax for things like links and inline code without the need for cumbersome markup.

giamir replied:
I am ok to merge this in but not having tests certainly makes me a bit uncomfortable about it. Especially because I see how this converter might need to evolve/change over time.

giamir replied:
Have you considered treating everything as markdown without having to add a dedicated shortcode?
Could we process all the markdown in a view via the converter by default?
Maybe we could start by adding a boolean in the front matter if we are afraid of side effects in other pages.

I am suggesting this because I don't particularly like constantly opening and closing shortcodes like this `{% stacks-markdown %} ... {% endstacks-markdown %}`. They also make syntax highlighting pretty poor compared to how it was before with just plain html (which had the problem of being cluttered with a lot of repeated tags and classes).

dancormier commented on docs/_includes/accessibility/comparison.html:
This is the markup for "comparison" blocks like this:

![image](https://github.com/StackExchange/Stacks/assets/647177/40ed679e-683c-434d-9fdf-8c55f2e87dc4)

dancormier committed:
Add comments explaining added/modified partials

dancormier committed:
add a little space

dancormier committed:
Clean up generateHeaderHTML function

dancormier commented on docs/product/base/color-fundamentals.html:
This file contains two types of changes:

- Rename `color-row-detail.html` to `example-row-detail.html`
- Update markup to use and support the newly generalize `color-swatch.html` partial

dancormier committed:
minor whitespace updates

dancormier committed:
minor a11y page tweaks

CGuindon commented:
@dancormier Yup! I have a note in the ticket (last bullet) but basically remove the [entire](https://deploy-preview-1670--stacks.netlify.app/product/base/color-fundamentals/#accessibility-standards) section from that page since it will now live on the new page. 

I also have some left nav hierarchy changes mentioned in the design specs — open to feedback on these.

dancormier committed:
Remove color-fundamentals "Accessibility standards" section

CGuindon reviewed (changes requested):
Did a quick visual pass over:

1. The link is correct but I just realized our copy had a typo, the HC link should for " Success Criterion 1.4.3 Contrast (Enhanced)" should "1.4.**6**"

<img width="825" alt="Screenshot 2024-03-15 at 1 58 25 PM" src="https://github.com/StackExchange/Stacks/assets/5436760/5e21bcae-91ad-48a1-88e6-46f050855208">
<img width="468" alt="Screenshot 2024-03-15 at 1 57 56 PM" src="https://github.com/StackExchange/Stacks/assets/5436760/a1cb4da1-22ba-4ee7-8741-b92f1cd73544">

---

2. Any chance we can easily get the color swatches to be vertically aligned in the middle? If it's too annoying, I can live with it. 
<img width="323" alt="Screenshot 2024-03-15 at 2 06 18 PM" src="https://github.com/StackExchange/Stacks/assets/5436760/5f361dc6-2ba6-4727-86a6-175d95ca3809">

---

3. I noticed the little text labels on the images disappear when I re-size the screen to small, is that on purpose/normal?
<img width="534" alt="Screenshot 2024-03-15 at 2 08 01 PM" src="https://github.com/StackExchange/Stacks/assets/5436760/bdc4e1c2-f7b0-4abb-b201-ddbd6d9a5a25">

---

4. Left nav changes and moving the Current Color section

giamir commented on docs/product/guidelines/accessibility.html:
**Sidenote (No action required)**

I look at something like this and I cannot stop but thinking how much more elegant/readable would be to use a Svelte component. This gives me the Razor vibe 😀

dancormier replied:
I absolutely agree. I would love to rearchitect our docs to use SvelteKit for reasons like that one (and many more)

giamir reviewed (commented):
Great work Dan. Looks good overall. 🎉

I am happy to merge it in although I am not entirely sold on the `stacks-markdown` plugin as it is now. I understand that it will save us from repeating classes and tags over and over but I dislike that we need to open and close shortcodes and we lose syntax highlighting in the process. I made a suggestion about exploring to run the converter you wrote against the whole markdown found in the view file without the usage of shortcodes (see review comment).

Also in my opinion we should make an effort to test that converter even if it's just for our docs. I can see how we want to change/add things there over time and having test cases will help us to do that with the confidence of not breaking things. Let me know if you need support setting those tests up (I would recommend to use vitest if you want to give it a shot - e.g. `"test:docs:unit": "vitest path/to/docs"`) 🙂

dancormier committed:
Remove stacks-markdown plugin from this branch

Addresses https://github.com/StackExchange/Stacks/pull/1670#pullrequestreview-1942358113

Note: this plugin has been moved to the branch `dcormier/docs-markdown-support` so we can iterate and consider it independently from the addition of the accessibility docs page.

dancormier commented:
> Great work Dan. Looks good overall. 🎉
> 
> I am happy to merge it in although I am not entirely sold on the `stacks-markdown` plugin as it is now. I understand that it will save us from repeating classes and tags over and over but I dislike that we need to open and close shortcodes and we lose syntax highlighting in the process. I made a suggestion about exploring to run the converter you wrote against the whole markdown found in the view file without the usage of shortcodes (see review comment).
> 
> Also in my opinion we should make an effort to test that converter even if it's just for our docs. I can see how we want to change/add things there over time and having test cases will help us to do that with the confidence of not breaking things. Let me know if you need support setting those tests up (I would recommend to use vitest if you want to give it a shot - e.g. `"test:docs:unit": "vitest path/to/docs"`) 🙂

You're the voice of reason here, @giamir. I've removed the `stacks-markdown` from this PR (in commit https://github.com/StackExchange/Stacks/pull/1670/commits/46e0cdbfdeac6d570625ef084a0b709c86938a10) so we can keep this PR scoped reasonably and get the accessibility page merged without hesitation.

FWIW, I think you're right to not be totally sold on the `stacks-markdown` plugin. I think we can explore it further when the time is right. For now, I've ported the plugin to the [dcormier/docs-markdown-support branch](https://github.com/StackExchange/Stacks/tree/dcormier/docs-markdown-support) so we have the code if/when we feel like revisiting.

Let me know if this seems reasonable.

dancormier committed:
Update the site navigation to match Figma mock

Partially addresses https://github.com/StackExchange/Stacks/pull/1670#pullrequestreview-1940639363

dancormier committed:
Fix typo (SC 1.4.3 ➡️ SC 1.4.6)

Partially addresses https://github.com/StackExchange/Stacks/pull/1670#pullrequestreview-1940639363

dancormier committed:
Center the color swatches

Partially addresses https://github.com/StackExchange/Stacks/pull/1670#pullrequestreview-1940639363

dancormier committed:
Show example detail in all viewport widths when possible

Partially addresses https://github.com/StackExchange/Stacks/pull/1670#pullrequestreview-1940639363

dancormier commented:
Thanks for the review @CGuindon. I've made changes that should address the issues you've found.

> Did a quick visual pass over:
> 
> 1. The link is correct but I just realized our copy had a typo, the HC link should for " Success Criterion 1.4.3 Contrast (Enhanced)" should "1.4.**6**"
> 
> <img alt="Screenshot 2024-03-15 at 1 58 25 PM" width="825" src="https://private-user-images.githubusercontent.com/5436760/313331997-5e21bcae-91ad-48a1-88e6-46f050855208.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NjA2ODgsIm5iZiI6MTcxMDg2MDM4OCwicGF0aCI6Ii81NDM2NzYwLzMxMzMzMTk5Ny01ZTIxYmNhZS05MWFkLTQ4YTEtODhlNi00NmYwNTA4NTUyMDgucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMTQ1OTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MGFmYzc2OTk5NTIwNTgzODJhOGQ3YTJmMTUwNjBhZTQwNzI1NWIyZTk0ZmI1ZTgyMzQ0Y2ZlY2UwOTQ5ZjE0ZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6_hPKHsoYjrfrROyvNRn2NesS1n97UJ3t8SC_USoGYI"> <img alt="Screenshot 2024-03-15 at 1 57 56 PM" width="468" src="https://private-user-images.githubusercontent.com/5436760/313332011-a1cb4da1-22ba-4ee7-8741-b92f1cd73544.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NjA2ODgsIm5iZiI6MTcxMDg2MDM4OCwicGF0aCI6Ii81NDM2NzYwLzMxMzMzMjAxMS1hMWNiNGRhMS0yMmJhLTRlZTctODc0MS1iOTJmMWNkNzM1NDQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMTQ1OTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2U3NDgwYThjYWU1NzIxMTJiYTE1MzUzYzBjZjQwMmFmN2EyMDU4MzY4OTAyNDk4ZGMxMjkyYWRlNDk4OGY5YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.PIc9tlphD_gvIhlGqIQRfRNth1ZyTrs5DyaNAfbd55U">

✅ Fixed in https://github.com/StackExchange/Stacks/pull/1670/commits/04c76bc9022d1b52c1d414e0f2f0ebdf7583c6b9

---

> 2. Any chance we can easily get the color swatches to be vertically aligned in the middle? If it's too annoying, I can live with it.
> 
> 
> <img alt="Screenshot 2024-03-15 at 2 06 18 PM" width="323" src="https://private-user-images.githubusercontent.com/5436760/313333688-5f361dc6-2ba6-4727-86a6-175d95ca3809.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NjA2ODgsIm5iZiI6MTcxMDg2MDM4OCwicGF0aCI6Ii81NDM2NzYwLzMxMzMzMzY4OC01ZjM2MWRjNi0yYmE2LTQ3MjctODZhNi0xNzVkOTVjYTM4MDkucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMTQ1OTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGJjZDdhNGRmZmVkYWMzZWQwYjU0NjM0M2E4OWFjMzQ4MzI4YmY2Y2JmMjc1NzI3Mjk3ZWQwNzJkOGZjMWEwNCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.bI3iz6BKIkjboHGXRK-uueLcbbOUOK-PUY3sfl6Qf20">

✅ Fixed in https://github.com/StackExchange/Stacks/pull/1670/commits/55ee4694d351e9c2511d75fb873cc9f064597ea2

---

> 3. I noticed the little text labels on the images disappear when I re-size the screen to small, is that on purpose/normal?
> 
> 
> <img alt="Screenshot 2024-03-15 at 2 08 01 PM" width="534" src="https://private-user-images.githubusercontent.com/5436760/313334399-bdc4e1c2-f7b0-4abb-b201-ddbd6d9a5a25.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTA4NjA2ODgsIm5iZiI6MTcxMDg2MDM4OCwicGF0aCI6Ii81NDM2NzYwLzMxMzMzNDM5OS1iZGM0ZTFjMi1mN2IwLTRhYmItYjIwMS1kZGJkNmQ5YTVhMjUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTlUMTQ1OTQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NmMxNzgxZWRmOTg2NDhkNzJjNThkYTQ4OTRmMTg1NDYyMjc3YTExMzI4MjFkMWU0YjU3YWVmMDgyY2UxMTJkNSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.ZNEhN5ZprnZ-1ZJlsv5lJzqMA_cz-CHSns_7XRntL54">

The disappearing labels are a remnant from the Color Fundamentals page build. Since the color rows in [the Usage section](https://stackoverflow.design/product/base/color-fundamentals/#backgrounds) shift to a vertical orientation in smaller viewports, it didn't make sense to show those labels in that partial. It also just kinda helped sidestep having to deal with ensuring the examples wrapped/flowed gracefully.

![image](https://github.com/StackExchange/Stacks/assets/647177/786c77aa-397c-409e-bfce-56b4da66f506)

![image](https://github.com/StackExchange/Stacks/assets/647177/1ee1a181-9671-451c-bbef-1e72852c82a5)

I've changed the partial to show those labels at all screen sizes unless we override it in https://github.com/StackExchange/Stacks/pull/1670/commits/087774b38a93c2a92dc5f88e366a896386fbc681. Let me know how this looks to you. I'm happy to make modifications if you think we should.

---

> 4. Left nav changes and moving the Current Color section

✅ Completed in https://github.com/StackExchange/Stacks/pull/1670/commits/e51495d3f7723d06d48ffea15290366e2424bc49

giamir commented:
> FWIW, I think you're right to not be totally sold on the stacks-markdown plugin. I think we can explore it further when the time is right. For now, I've ported the plugin to the [dcormier/docs-markdown-support branch](https://github.com/StackExchange/Stacks/tree/dcormier/docs-markdown-support) so we have the code if/when we feel like revisiting.

It works for me. Maybe you can also create a quick companion "operational maintenance" jira ticket if it makes sense (it will probably end up in the low priority pile but at least we don't lose track of the improvement). Thank you. 😊

giamir commented:
@dancormier @CGuindon Is the Accessibility navigation item under the correct category?
https://deploy-preview-1670--stacks.netlify.app/product/guidelines/accessibility/

<img width="1272" alt="Screenshot 2024-03-20 at 09 53 58" src="https://github.com/StackExchange/Stacks/assets/6231616/c2fb8b61-e084-4c48-8e2e-80ce9d7ed404">

dancormier commented:
> @dancormier @CGuindon Is the Accessibility navigation item under the correct category? https://deploy-preview-1670--stacks.netlify.app/product/guidelines/accessibility/

Yep, I originally missed the reorganization of that navigation in the [Figma](https://www.figma.com/file/3OlPL6x0chNbMIPjP2vxas/Stack-Overflow-3.0---Site-Modernization?node-id=3522%3A4090&mode=dev) and made those changes in https://github.com/StackExchange/Stacks/commit/e51495d3f7723d06d48ffea15290366e2424bc49

CGuindon commented:
> > @dancormier @CGuindon Is the Accessibility navigation item under the correct category? https://deploy-preview-1670--stacks.netlify.app/product/guidelines/accessibility/
> 
> Yep, I originally missed the reorganization of that navigation in the [Figma](https://www.figma.com/file/3OlPL6x0chNbMIPjP2vxas/Stack-Overflow-3.0---Site-Modernization?node-id=3522%3A4090&mode=dev) and made those changes in [e51495d](https://github.com/StackExchange/Stacks/commit/e51495d3f7723d06d48ffea15290366e2424bc49)

@giamir @dancormier I also wanted to change the name of that top section to "Develop" — since everything in here is a _Guideline_. Thoughts or concerns?

<img width="239" alt="Screenshot 2024-03-20 at 3 58 07 PM" src="https://github.com/StackExchange/Stacks/assets/5436760/7bcdce91-d80c-44ee-b524-f1c6a6aa2d3e">

CGuindon reviewed (changes requested):
Commented on this [commit](https://github.com/StackExchange/Stacks/commit/55ee4694d351e9c2511d75fb873cc9f064597ea2) directly since I can't push any changes (permission thing I think?). Otherwise the other changes look good.

dancormier committed:
Tweak color swatch bottom margin

Addresses https://github.com/StackExchange/Stacks/commit/55ee4694d351e9c2511d75fb873cc9f064597ea2#commitcomment-140046035

dancormier committed:
Fix site navigation titles

Addresses https://github.com/StackExchange/Stacks/pull/1670#issuecomment-2010840773

dancormier commented:
### [Color swatch bottom margin](https://github.com/StackExchange/Stacks/pull/1670#pullrequestreview-1950425284)

@CGuindon Updated!

> Commented on this [commit](https://github.com/StackExchange/Stacks/commit/55ee4694d351e9c2511d75fb873cc9f064597ea2) directly since I can't push any changes (permission thing I think?). Otherwise the other changes look good.

### [Docs navigation changes](https://github.com/StackExchange/Stacks/pull/1670#issuecomment-2010840773)

> I also wanted to change the name of that top section to "Develop" — since everything in here is a Guideline. Thoughts or concerns?

@CGuindon I think that'll work fine. I originally changed the wrong heading but have fixed it in https://github.com/StackExchange/Stacks/pull/1670/commits/d092389d26a14810a4e2fa89070c2159dd723641

I made [another PR](https://github.com/StackExchange/Stacks/pull/1684) to update urls to better reflect the nav structure. This isn't _entirely_ necessary but it feels like a good excuse to perform that cleanup. @giamir @CGuindon when you get a moment, let me know what you think of that PR and I'll either merge it into this PR or close it out.

dancormier committed:
docs(navigation): remap paths to reflect nav sections (#1684)

* docs(navigation): remap paths to reflect nav sections

* chore(redirects): update existing redirects to new paths

Addresses https://github.com/StackExchange/Stacks/pull/1684#pullrequestreview-1954502163 and https://github.com/StackExchange/Stacks/pull/1684#discussion_r1535395068

* Update test utils comment docs link path

Addresses https://github.com/StackExchange/Stacks/pull/1684#discussion_r1535398814
